pharus/component_interface.py

```diff
@@ -74,19 +73,39 @@ def __init__(self, name, component_config, static_config, jwt_payload: dict):
         else:
             self.dj_restriction = lambda: dict()
 
-        self.vm_list = [
-            dj.VirtualModule(
-                s,
-                s,
-                connection=dj.conn(
-                    host=jwt_payload["databaseAddress"],
-                    user=jwt_payload["username"],
-                    password=jwt_payload["password"],
-                    reset=True,
-                ),
-            )
-            for s in inspect.getfullargspec(self.dj_query).args
-        ]
+        if "dj_query" in component_config:
```

Comment:
Think it would be best to split into a separate fundamental component. We can perhaps rename this one to be `FetchComponent` and create a new base `InsertComponent`.